src/engine/Events.ts

```diff
@@ -545,3 +549,21 @@ export class ExitTriggerEvent extends GameEvent<Actor> {
     super();
   }
 }
+
+/**
+ * Event thrown on an [[Actor]] when an action starts.
+ */
+export class ActionStartEvent extends GameEvent<Entity> {
```

Comment:
@eonarheim I wasn't too sure if this should be `GameEvent<Entity>` or if you'd have wanted an interface to be made so I could do the `<T extends XYZ = Entity>` trick?

Reply:
@mattjennings This is perfect for now! Fits with the current pattern. 

Eventually I want to rethink GameEvent entirely (maybe remove it?) but I haven't put enough effort into what the future would look like.

Reply:
sounds good! 

It's pretty nice having these to use as types when setting up handlers on a class:

```ts
onInitialize() {
     this.on('actionstart', this.onActionStart)
}

onActionStart(ev: ex.ActionStartEvent) { ... }
```

so I'd advocate for keeping them